packages/editor/src/components/rich-text/index.js

```diff
@@ -136,7 +136,17 @@ export class RichText extends Component {
 	}
 
 	setRef( node ) {
-		this.editableRef = node;
+		if ( node ) {
+			const computedStyle = getComputedStyle( node );
+
+			if ( computedStyle.display === 'inline' ) {
+				window.console.warn( 'RichText cannot be used with an inline container. Please use a different tagName.' );
```

Comment:
Is the use of `window.console.warn` is to get around the lint rule? In other places I noticed the rule is specifically disabled:
```
// eslint-disable-next-line no-console
console.warn( // ... );
```
I think it'd be better to follow that convention—if the eslint rule is updated in the future to cover `window.console` there would lint failures that need to be fixed.

Reply:
> if the eslint rule is updated in the future to cover `window.console` there would lint failures that need to be fixed.

This conversation makes me want to update that rule now 😅

Reply:
I wonder, now that we add the environment check, if we cannot just stop `RichText` from doing anything further. It will still work in production, but if you're trying to develop with RichText and inline elements, it just won't work. I think a lot of people will just ignore or be blind to the warning.